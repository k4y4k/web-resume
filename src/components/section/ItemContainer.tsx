import * as React from 'react'
import Title from './ItemTitle'
import Subtitle from './ItemSubtitle'
import Dates from './ItemDates'
import Details from './ItemDetails'

interface ItemContainerTypes {
  title?: string
  subtitle?: string
  fromDate?: string
  toDate?: string
  details?: string
}

const ItemContainer = ({
  title,
  subtitle,
  fromDate,
  toDate,
  details,
}: ItemContainerTypes): JSX.Element => (
  <>
    <Title title={title} />
    <Subtitle subtitle={subtitle} />
    <Dates from={fromDate} to={toDate} />
    <Details details={details} />
  </>
)

export default ItemContainer
