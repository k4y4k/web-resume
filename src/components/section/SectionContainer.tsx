import 'twin.macro'
import * as React from 'react'
import SectionTitle from './SectionTitle'

interface SectionContainerTypes {
  title: string
  children: React.ReactNode
}

const SectionContainer = ({
  title,
  children,
}: SectionContainerTypes): JSX.Element => (
  <div tw='p-4'>
    <SectionTitle title={title} />
    {children}
  </div>
)

export default SectionContainer
