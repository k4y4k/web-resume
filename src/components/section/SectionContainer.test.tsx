import * as React from "react";
import { render, screen } from "@testing-library/react";
import ItemContainer from "./ItemContainer";
import SectionContainer from "./SectionContainer";

describe("<SectionContainer />", () => {
  test("composes title and >1  items", () => {
    const data = (
      <ItemContainer title="" subtitle="" fromDate="" toDate="" summary="" />
    );

    render(<SectionContainer children={data} title="Reason #8: Regret" />);

    const title = screen.getByText("Reason #8: Regret");
    expect(title).toBeInTheDocument();

    const item = screen.getAllByTestId("sectionItemContainer");
    expect(item.length).toBeGreaterThanOrEqual(1);

    expect(screen.getByTestId("sectionContainer")).toMatchSnapshot();
  });
});
