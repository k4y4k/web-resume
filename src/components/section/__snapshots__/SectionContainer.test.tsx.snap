// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> composes title and >1  items 1`] = `
.c2 {
  margin-top: 1rem;
  margin-bottom: 1rem;
}

.c2:nth-of-type(1) {
  margin-top: 0.5rem;
}

.c3 {
  font-style: italic;
}

.c1 {
  font-family: Inconsolata,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;
  font-size: 1.25rem;
  line-height: 1.75rem;
  font-weight: 700;
  text-transform: uppercase;
  --tw-text-opacity: 1;
  color: rgba(77,54,140,var(--tw-text-opacity));
}

.c0 {
  padding: 1rem;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
}

<div
  class="c0"
  data-testid="sectionContainer"
>
  <h2
    class="c1"
    data-testid="sectionTitle"
  >
    Reason #8: Regret
  </h2>
  <div
    class="c2"
    data-testid="sectionItemContainer"
  >
    <div
      class="c3"
      data-testid="byline"
    />
  </div>
</div>
`;
